services/secondary-storage/server/src/org/apache/cloudstack/storage/template/DownloadManagerImpl.java

```diff
@@ -984,7 +984,8 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
         String inSystemVM = (String)params.get("secondary.storage.vm");
         if (inSystemVM != null && "true".equalsIgnoreCase(inSystemVM)) {
             s_logger.info("DownloadManager: starting additional services since we are inside system vm");
-            _nfsVersion = (String)params.get("nfsVersion");
+            String nfsVersionParam = (String)params.get("nfsVersion");
+            _nfsVersion = (nfsVersionParam != null ? Integer.valueOf(nfsVersionParam) : null);
```

Comment:
Saw this pattern getting repeated in multiple places, if possible move it to a method.